### PR TITLE
fix for missing thread ID in stop reply when smp-configured harts other than hart 0 single-stepped

### PR DESCRIPTION
**Pull request by GregSavin**
To replicate the issue that this fixes:

1) checkout the riscv branch, build, connect to a multi-hart target configured as an SMP group.  
2) Start a GDB instance against the running OpenOCD.  
3) Observe that GDB might display "warning: multi-threaded target stopped without sending a thread-id, using first non-exited thread."
4) Set a breakpoint in code that any non-hart-0 hart is expected to reach (but hart 0 is not expected to reach).
5)  Allow a non-hart-0 hart to reach the breakpoint.
6) Remove the breakpoint.
7) Do a few sequential "stepi" commands in GDB.
8) Observe that GDB displays "Switching to Thread 1" even though the thread that was just single stepped was not Thread 1 in GDB.  Also observe that the register values in GDB correspond to the thread that was single-stepped, not Thread 1.  Basically GDB erroneously starts to consider thread 1 to be current, when in fact the thread that was single-stepped is still current.

The changes in this pull request are intended to avoid the erroneous "Switching to Thread 1" described in (8) above.

What was happening was that, in a couple areas of code, non-hart-0 harts weren't seen as belonging to an rtos module, and this had the effect of (1) bypassing hwthread_update_threads() being called after a halt; (2) omitting a thread ID in a stop reply over GDB remote protocol connection (requiring GDB to take an arbitrary guess of current thread id, a guess that is wrong unless the current thread happens to be hart 0).

### CODE REVIEW

**Comment by GregSavin** [comment]
I ran riscv-tests/debug/make on this branch, and didn't see any failing tests.  Note that there is not yet a test in riscv-tests/debug that covers this specific current issue.

**Comment by timsifive** [comment]
> I ran riscv-tests/debug/make on this branch, and didn't see any failing tests. Note that there is not yet a test in riscv-tests/debug that covers this specific current issue.

Did you run `make` or `make all`? The latter runs more tests, and covers a few more corner cases that happen in different multicore setups.

File: src/rtos/rtos.c
Comment (timsifive) [inline src/rtos/rtos.c]: Something is wonky with the indenting here.

**Review by timsifive — DISMISSED** [review]
The change you made is definitely right for targets configured with SMP and `-rtos hwthread`. I'm not sure it's right for every combination of those options, but I think that the intent behind those two features is to be used together. E.g. while technically you could have a target with SMP cores and ThreadX running on only one of them, that is not the intended use case behind the features. I suspect you're going to have a hard time with that in gdb regardless.

So I think this change is good.

**Comment by GregSavin** [comment]
> Did you run `make` or `make all`? The latter runs more tests, and covers a few more corner cases that happen in different multicore setups.

I ran "make".  I'll check again with "make all"

**Comment by GregSavin** [comment]
@timsifive "make all" looked OK.

[riscv-tests/debug]$ make all
(voluminous output not shown)
[riscv-tests/debug]$ grep "Result: pass" logs/* | wc -l
323
[riscv-tests/debug]$ grep "Result: fail" logs/* | wc -l
0
[riscv-tests/debug]$ grep "Result: exception" logs/* | wc -l
0
[riscv-tests/debug]$ grep "Result: not_applicable" logs/* | wc -l
52

File: src/rtos/rtos.c
Comment (GregSavin) [reply src/rtos/rtos.c]: Latest revision fixes the indentation.  Looks like my commit had the side-effect of github invalidating your prior approval, so I guess this needs an approval again, before I'm allowed to press the merge button.

**Comment by GregSavin** [comment]
@timsifive Do you think it'd be worthwhile if I add a new test case to riscv-tests to get regression coverage for the scenario that this patch is a fix for?

**Comment by timsifive** [comment]
> @timsifive Do you think it'd be worthwhile if I add a new test case to riscv-tests to get regression coverage for the scenario that this patch is a fix for?

I think that'd be great. I really should have made a testcase for the software breakpoint with different harts having different address translation, but I'm busy, <insert worthless excuse>. Having coverage for this kind of thing is important because it's exactly the kind of thing that could easily break while making some other improvement.

**Review by timsifive — APPROVED** [review]

**Pull request merged**
